integration/indexes_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+)
+
+func TestIndexesDropRunCommandErrors(t *testing.T) {
+	setup.SkipForTigrisWithReason(t, "Indexes are not supported for Tigris")
+
+	t.Parallel()
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		toCreate []mongo.IndexModel // optional, if set, create the given indexes before drop is called
+		toDrop   any                // required, index to drop
+		command  bson.D             // optional, if set it runs this command instead of dropping toDrop
+
+		err        mongo.CommandError // required
+		altMessage string             // optional, alternative error message
+		skip       string             // optional, skip test with a specified reason
+	}{
+		"InvalidType": {
+			toDrop: true,
+			err: mongo.CommandError{
+				Code:    14,
+				Name:    "TypeMismatch",
+				Message: "BSON field 'dropIndexes.index' is the wrong type 'bool', expected types '[string, object']",
+			},
+			altMessage: `BSON field 'dropIndexes.index' is the wrong type 'bool', expected types '[string, object]'`,
+		},
+		"MultipleIndexesByKey": {
+			toCreate: []mongo.IndexModel{
+				{Keys: bson.D{{"v", -1}}},
+				{Keys: bson.D{{"v.foo", -1}}},
+			},
+			toDrop: bson.A{bson.D{{"v", -1}}, bson.D{{"v.foo", -1}}},
+			err: mongo.CommandError{
+				Code:    14,
+				Name:    "TypeMismatch",
+				Message: "BSON field 'dropIndexes.index' is the wrong type 'array', expected types '[string']",
+			},
+			altMessage: `BSON field 'dropIndexes.index' is the wrong type 'array', expected types '[string, object]'`,
```

Comment:
That's weird. Mongo expects `string` but we expect `string` and `object`. @rumyantseva maybe you if this is the correct behaviour?